### PR TITLE
YDA-7171: fix publication handling affiliations

### PR DESCRIPTION
**Pull request by stsnel**
Fix handling of affiliations without an affiliation name in the publication functions. Affiliations without a name are accepted by the current metadata schemas, so the publication functions should support this too.

### CODE REVIEW

File: json_datacite.py [hunk 1]
```diff
@@ -165,6 +165,31 @@ def get_funders(combi: dict) -> List:
     return funders
 
 
+def _process_affiliations_list(input: List) -> List[dict]:
```
Comment (leonidastri) [inline json_datacite.py]: should we change input as it shadows the Python builtin?

File: json_datacite.py [hunk 2]
```diff
@@ -165,6 +165,31 @@ def get_funders(combi: dict) -> List:
...
+    :param input: List of affiliations in Yoda metadata format
+
+    :returns: List of dictionaries with affiliation data in DataCite format
+    """
+    affiliations: List[dict] = []
+    for aff in input:
+        affiliation_data = {}
+        if isinstance(aff, dict):
+            if "Affiliation_Name" in aff and len(aff["Affiliation_Name"]):
+                affiliation_data["name"] = aff['Affiliation_Name']
```
Comment (leonidastri) [inline json_datacite.py]: in the old code we used to transform it to string. is it guaranteed that to be a string by the schema?

File: json_datacite.py (hunk 1 above)
Comment (leonidastri) [inline json_datacite.py]: could there be an edge case causing a type error in this function, which could cause a crash on the publication?

**Comment by leonidastri** [comment]
Clear implementation and easy to understand, great test coverage.  I have added some minor comments.

File: json_datacite.py (hunk 2 above)
Comment (stsnel) [reply json_datacite.py]: Thanks, have re-added the str conversion to ensure who don't change behaviour.

File: json_datacite.py (hunk 1 above)
Comment (stsnel) [reply json_datacite.py]: I've tried to incorporate the relevant edge cases in the unit tests. Do you have any suggestions for edge cases that still need to be covered?

Comment (leonidastri) [reply json_datacite.py]: I think the test coverage is good for now. We can always extend the test cases later.

**Review by leonidastri — APPROVED** [review]
LGTM!
